tests/syscall/mount/mount-immutables-unshare.bats

```diff
@@ -0,0 +1,663 @@
+  run empty_list ${immutable_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in $immutable_mounts; do
+    # Skip /proc and /sys since these are special mounts (we have dedicated
+    # tests that cover unmounting ops).
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+        [[ ${m} =~ "/sys" ]] || [[ ${m} =~ "/sys/*" ]]; then
+        continue
+    fi
+
+    printf "\ntesting unmount of immutable mount ${m}\n"
+
+    docker exec ${syscont} sh -c "nsenter -a -t ${inner_pid} umount ${m}"
+    [ "$status" -ne 0 ]
+  done
+
+  local immutable_mounts_after=$(list_container_mounts ${syscont} ${inner_pid} "/")
+  [[ $immutable_mounts == $immutable_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #2.
+#
+# Ensure that a read-only immutable mount can't be remounted as read-write
+# inside an inner mount namespace.
+@test "immutable ro mount can't be remounted rw -- unshare(mnt)" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  
+  docker exec -d ${syscont} sh -c "unshare -m bash -c \"sleep 1000\""
+  [ "$status" -eq 0 ]
+
+  docker exec ${syscont} sh -c "pidof sleep"
+  [ "$status" -eq 0 ]
+  local inner_pid=$output
+
+  local immutable_ro_mounts=$(list_container_ro_mounts ${syscont} ${inner_pid} "/")
+  run empty_list ${immutable_ro_mounts}
+  [ "$status" -ne 0 ]
+
+  for m in ${immutable_ro_mounts}; do
+    printf "\ntesting rw remount of immutable ro mount ${m}\n"
+
+    docker exec ${syscont} sh -c "nsenter -a -t ${inner_pid} mount -o remount,bind,rw ${m}"
+    [ "$status" -ne 0 ]
+  done
+
+  local immutable_ro_mounts_after=$(list_container_ro_mounts ${syscont} ${inner_pid} "/")
+  [[ $immutable_ro_mounts == $immutable_ro_mounts_after ]]
+
+  docker_stop ${syscont}
+}
+
+# Testcase #3.
+#
+# Ensure that a read-write immutable mount *can* be remounted as read-only inside
+# an inner mount namespace, and then back to read-only.
+@test "immutable rw mount can be remounted ro -- unshare(mnt)" {
+  if [[ $skipTest -eq 1 ]]; then
+    skip
+  fi
+  local syscont=$(docker_run --rm ${CTR_IMG_REPO}/ubuntu:latest tail -f /dev/null)
+  
+  docker exec -d ${syscont} sh -c "unshare -m bash -c \"sleep 1000\""
+  [ "$status" -eq 0 ]
+
+  docker exec ${syscont} sh -c "pidof sleep"
+  [ "$status" -eq 0 ]
+  local inner_pid=$output
+
+  local immutable_rw_mounts=$(list_container_rw_mounts ${syscont} ${inner_pid} "/")
+  run empty_list ${immutable_rw_mounts}
+  [ "$status" -ne 0 ]  
+
+  for m in ${immutable_rw_mounts}; do
+
+    # Remounting /proc or /dev as read-only will prevent docker execs into the
+    # container; skip these.
+    if [[ ${m} =~ "/proc" ]] || [[ ${m} =~ "/proc/*" ]] ||
+       [[ ${m} =~ "/dev" ]] || [[ ${m} =~ "/dev/*" ]]; then
+      continue
+    fi
+
+    printf "\ntesting ro remount of immutable rw mount ${m}\n"
+
+    docker exec ${syscont} sh -c "nsenter -a -t ${inner_pid} mount -o remount,bind,ro ${m}"
+    [ "$status" -eq 0 ]
```

Comment:
Suggestion: add "touch" after each mount step to verify the mount is read-write or read-only.